python_modules/libraries/dagster-dbt/dagster_dbt/dbt_manifest_asset_selection.py

```diff
@@ -73,3 +74,28 @@ def resolve_inner(self, asset_graph: AssetGraph) -> AbstractSet[AssetKey]:
                 keys.add(asset_key)
 
         return keys
+
+    def resolve_checks_inner(self, asset_graph: AssetGraph) -> AbstractSet[AssetKey]:
+        if not self.dagster_dbt_translator.settings.enable_asset_checks:
+            return set()
+
+        dbt_nodes = get_dbt_resource_props_by_dbt_unique_id_from_manifest(self.manifest)
+
+        keys = set()
+        for unique_id in select_unique_ids_from_manifest(
+            select=self.select,
+            exclude=self.exclude,
+            manifest_json=self.manifest,
+        ):
+            dbt_resource_props = dbt_nodes[unique_id]
+            if dbt_resource_props["resource_type"] == "test":
+                attached_node_unique_id = dbt_resource_props.get("attached_node")
+                is_generic_test = bool(attached_node_unique_id)
+
+                if is_generic_test:
```

Comment:
nit: invert the condition logic and use `continue` so we don't get these nested if statements